pkg/markdown/parser/document.go

```diff
@@ -9,12 +9,27 @@ type document struct {
 	links []Link
 }
 
-func (d *document) ListLinks(cb UpdateMarkdownLinkListed) {
+func (d *document) ListLinks(cb UpdateMarkdownLinkListed) error {
```

Comment:
I think ListLinks is not a suitable name anymore as it is updating the list. We should somehow suggest that if the callback returns nil for the link it will be removed from the list. What do you think

Reply:
No, it still has the dedicated purpose to iterate links (maybe list should change to something more in the lines of iterate?). Note that it is `UpdateMarkdownLinks` (hence the name) that does changes. If you check its implementation you will notice that `UpdateMarkdownLinks` uses `ListLinks` to iterate on links and delegate link processing to a callback as a closure.